### ISSUE TITLE
Fail fast approach when running multiple test functions in a test package

### ISSUE THREAD

**Issue by Madzee** [kubernetes-sigs/e2e-framework#99]
Provide a flag during test execution to  exit execution on the occurrence of first test failure. 
Example : If the test package had 5 test functions, and the 3rd test function were to fail, the flag would allow the execution to exit at this point without running the remaining tests.

**Comment by vladimirvivien** [comment]
@Madzee I think this is similar to issue #98 

This is definitely something that would be useful.

**Comment by harshanarayana** [comment]
xref: #112

**Comment by vladimirvivien** [comment]
I think this can be closed.

**Comment by harshanarayana** [comment]
Yes. We should close this one. The `--fail-fast` takes are of the ask from this one.
